### DISCUSSION TITLE
How to read a string?

### DISCUSSION THREAD

**Discussion by ritulahkar** [Rahix/avr-hal#437] [Q&A]
Is there any way to modify this ``nb::block!(serial.read()).void_unwrap()`` to read string. Or How to call String struct. Please help. I am a newbie.

**Comment by Rahix** [comment]
(Converted the issue to a discussion)

> Or How to call String struct.

`String` is not available on `#![no_std]` Rust because it requires an allocator.  You can use the [`heapless::String`](https://docs.rs/heapless/0.7.16/heapless/struct.String.html) type instead. It is backed by a fixed buffer instead of dynamically allocating memory at runtime.

You will then need to receive individual characters and write them into this string.
